src/test/sql/sql_translator_test.cpp

```diff
@@ -498,6 +498,34 @@ TEST_F(SQLTranslatorTest, Distinct) {
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
 }
 
+TEST_F(SQLTranslatorTest, DistinctAndGroupBy) {
+  const auto actual_lqp = compile_query("SELECT DISTINCT b FROM int_float GROUP BY b");
+
+  // clang-format off
+  const auto expected_lqp =
+  AggregateNode::make(expression_vector(int_float_b), expression_vector(),
+    stored_table_node_int_float);
+  // clang-format on
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+TEST_F(SQLTranslatorTest, AggregateWithDistinctAndGroupBy) {
+  const auto actual_lqp = compile_query("SELECT DISTINCT b, SUM(a * 3) * b FROM int_float GROUP BY b");
```

Comment:
If the query was `SELECT DISTINCT MIN(a), MAX(a) ...` there should be a second AggregateNode here, grouping by `MIN(a), MAX(a)` to make sure the tuples are unique, right?

Reply:
(I've never used SELECT DISTINCT, just guessing here)

Reply:
Ah, good point. I would have missed that.

Reply:
No, wait, could you give a full SQL query for your example? I now believe that the problem does not exist.

Reply:
Take `SELECT MIN(a) FROM t GROUP BY b`: the result could easily be

```
MIN(a)
int
3
5
5
9
```

I would expect the result of `SELECT MIN(a) FROM t GROUP BY b` to be 

```
MIN(a)
int
3
5
9
```

Reply:
You're right, thanks for catching this. Should be fixed now.